sermant-backend/src/main/java/io/sermant/backend/controller/ConfigController.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sermant.backend.controller;
+
+import io.sermant.backend.entity.config.ConfigInfo;
+import io.sermant.backend.entity.config.ConfigServerInfo;
+import io.sermant.backend.entity.config.PluginType;
+import io.sermant.backend.entity.config.Result;
+import io.sermant.backend.entity.config.ResultCodeType;
+import io.sermant.backend.service.ConfigService;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Resource;
+
+/**
+ * Config Controller
+ *
+ * @author zhp
+ * @since 2024-05-16
+ */
+@Component
+@RestController
+@RequestMapping("/sermant")
+public class ConfigController {
+    @Resource
+    private ConfigService configService;
+
+    /**
+     * Query Configuration List
+     *
+     * @param configInfo Query criteria for configuration list
+     * @return Configuration List
+     */
+    @GetMapping("/configs")
+    public Result<List<ConfigInfo>> getConfigList(ConfigInfo configInfo) {
+        if (StringUtils.isEmpty(configInfo.getPluginType())) {
+            return new Result<>(ResultCodeType.MISS_PARAM.getCode(), ResultCodeType.MISS_PARAM.getMessage());
+        }
+        Optional<PluginType> optionalPluginType = PluginType.getPluginType(configInfo.getPluginType());
+        if (!optionalPluginType.isPresent()) {
+            return new Result<>(ResultCodeType.FAIL.getCode(), "Invalid plugin name.");
+        }
+        PluginType pluginType = optionalPluginType.get();
+        if (pluginType == PluginType.OTHER) {
+            if (StringUtils.isEmpty(configInfo.getGroup())) {
```

Comment:
merge these two conditional statements

Reply:
When pluginType == PluginType.OTHER, logic processing is added and no combination is required.